tests/xharness/TestTasks/RunTest.cs

```diff
@@ -36,6 +39,7 @@ public class RunTest {
 			this.testTask = testTask ?? throw new ArgumentNullException (nameof (testTask));
 			this.BuildTask = buildTask ?? throw new ArgumentNullException (nameof (buildTask));
 			this.ProcessManager = processManager ?? throw new ArgumentNullException (nameof (processManager));
+			this.TunnelBore = tunnelBore; // can be null since some tests do not need tunnel
```

Comment:
there's a lot of other places where there's a `null` check...
I think this and `UseTcpTunnel` should be merged, e.g. `UseTcpTunnel` is only a getter that checks if the `TunnelBore` is non-null.